packages/nextjs/components/scaffold-eth/Balance.tsx

```diff
@@ -1,18 +1,25 @@
+import { useState } from "react";
 import { Address } from "viem";
 import { useAccountBalance } from "~~/hooks/scaffold-eth";
 import { useTargetNetwork } from "~~/hooks/scaffold-eth/useTargetNetwork";
 
 type BalanceProps = {
   address?: Address;
   className?: string;
+  usdMode?: boolean;
 };
 
 /**
  * Display (ETH & USD) balance of an ETH address.
  */
-export const Balance = ({ address, className = "" }: BalanceProps) => {
+export const Balance = ({ address, className = "", usdMode }: BalanceProps) => {
   const { targetNetwork } = useTargetNetwork();
-  const { balance, price, isError, isLoading, onToggleBalance, isEthBalance } = useAccountBalance(address);
+  const { balance, price, isError, isLoading, isEthBalance } = useAccountBalance(address);
+  const [displayUsdMode, setDisplayUsdMode] = useState<boolean | undefined>(usdMode);
+
+  const toggleBalanceMode = () => {
+    setDisplayUsdMode(prevMode => !prevMode);
+  };
```

Comment:
since its used only one time maybe we could just inline `setDisplayUsdMode` inside `onClick`